docs/deploying.md

```diff
@@ -1,35 +1,356 @@
+3. Start a registry service on your localhost.
 
-Note: Cloudfront keys exist separately to other AWS keys.  See [here](http://docs.aws.amazon.com/AWSSecurityCredentials/1.0/AboutAWSCredentials.html#KeyPairs) for more information.
+		$ docker run -p 5000:5000 registry
+		
+	This starts a registry on your `DOCKER_HOST` running on port `5000`. 
+	
+3. List your images.
+	
+		 $ docker images
+		 REPOSITORY     TAG     IMAGE ID      CREATED       VIRTUAL SIZE
+		 registry       2.0     bbf0b6ffe923  3 days ago    545.1 MB
+		 golang         1.4     121a93c90463  5 days ago    514.9 MB
+		 hello-world    latest  e45a5af57b00  3 months ago  910 B
+  
+    Your list should include a `hello-world` image from the earlier run.
 
-## Example
+4. Retag the `hello-world` image for your local repoistory.
 
+		$ docker tag hello-world:latest localhost:5000/hello-mine:latest
 
+	 The command labels a `hello-world:latest` using a new tag in the
+	 `[REGISTRYHOST/]NAME[:TAG]` format.  The `REGISTRYHOST` is this case is
+	 `localhost`. In a Mac OSX environment, you'd substitute `$(boot2docker
+	 ip):5000` for the `localhost`.
+	
+5. List your new image.
+
+		 $ docker images
+		 REPOSITORY                  TAG          IMAGE ID      CREATED       VIRTUAL SIZE
+		 registry                    2.0     bbf0b6ffe923  3 days ago    545.1 MB
+		 golang                      1.4     121a93c90463  5 days ago    514.9 MB
+		 hello-world                 latest  e45a5af57b00  3 months ago  910 B		 
+		 localhost:5000/hello-mine   latest  ef5a5gf57b01  3 months ago  910 B
+	
+	 You should see your new image in your listing.
+
+5. Push this new image to your local registry.
+
+		$ docker push localhost:5000/hello-mine:latest
+			
+6. Remove all the unused images from your local environment:
+
+		$ docker rmi -f $(docker images -q -a )
+
+	This command is for illustrative purposes; removing the image forces any `run`
+	to pull from a registry rather than a local cache. If you run `docker images`
+	after this you should not see any instance of `hello-world` or `hello-mine` in
+	your images list.
+	
+		 $ docker images
+		 REPOSITORY      TAG      IMAGE ID      CREATED       VIRTUAL SIZE
+		 registry         2.0     bbf0b6ffe923  3 days ago    545.1 MB
+		 golang           1.4     121a93c90463  5 days ago    514.9 MB
+	
+7. Try running `hello-mine`.
+
+		$ docker run hello-mine
+		Unable to find image 'hello-mine:latest' locally
+		Pulling repository hello-mine
+		FATA[0001] Error: image library/hello-mine:latest not found 
+		
+	The `run` command fails because your new image doesn't exist in the Docker public
+	registry.
+
+8. Now, try running the image but specifying the image's registry:
+
+		$ docker run localhost:5000/hello-mine
+
+	If you run `docker images` after this you'll fine a `hello-mine` instance. 
+		
+### The official image is not production ready
+
+Docker's official image is for simple tests or debugging. Its configuration is
```

Comment:
Let's spin this section: "Making the official image production ready". This could over or link the following:

1. Setting up backend.
2. Setting up access/auth.
3. Debug server.
4. Caching.

Then, we should how to override the configuration.